test/videojs-hls_test.js

```diff
@@ -1296,6 +1296,26 @@ test('blacklists switching from video-only playlists to video+audio', function()
   equal(videoAudioPlaylist.excludeUntil, Infinity, 'excluded incompatible playlist');
 });
 
+test('Switches to HAVE_METADATA if 404', function() {
```

Comment:
Couple of comments:

The goal of the test should be to ensure that `loadedmetadata` event is triggered on a successful load after a 404 for the first playlist. You should bind an event listener to `loadedmetadata` and record the number of times it was called. After the 404 it should have been called `0` times and after the successful playlist load it should be `1`.